tools/firtool/firtool.cpp

```diff
@@ -558,6 +558,9 @@ processBuffer(MLIRContext &context, TimingScope &ts, llvm::SourceMgr &sourceMgr,
     }
   }
 
+  pm.nest<firrtl::CircuitOp>().nest<firrtl::FModuleOp>().addPass(
+      firrtl::createRandomizeRegisterInitPass());
+
```

Comment:
This looks like a fine place to run it. The only optimizations before this shouldn't remove registers: (1) CSE and (2) Dedup.

Reply:
👍  yes I brainstormed with @uenoku and this seemed like a good spot.